scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -26,9 +26,27 @@ namespace <%= namespace %> {
             }
         }
 
-        private CartLineItems _LineItems;
+        private bool IsSaved {
+            get {
+                return IsCreated && 
+                    LineItems.New().Count == 0 &&
+                    LineItems.Modified().Count == 0 &&
+                    LineItems.Deleted().Count == 0;
+            }
+        }
+
+        private bool IsCreated {
+            get {
+                return CheckoutId != null;
+            }
+        }
 
+        private const float POLL_DELAY = 0.5f;
+        private CartLineItems _LineItems;
         private ShopifyClient Client;
+        private string CheckoutId = null;
```

Comment:
I dont really get why the cart model knows so much about the checkout, is this an attempt to merge these two concepts into one?